lib/MockXMLHttpRequest.js

```diff
@@ -158,40 +159,45 @@ MockXMLHttpRequest.prototype.send = function(data) {
 
     var response = MockXMLHttpRequest.handle(new MockRequest(self));
 
-    if (response && response instanceof MockResponse) {
+    Promise.resolve(response).then(function(response) {
```

Comment:
I'm not sure that this will handle the case where a promise has been returned by a handler without a response i.e. the handler did something async but still couldn't handle the response. 

Should it handle that case?